flight/Libraries/math/statistics.c

```diff
@@ -0,0 +1,352 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsMath Tau Labs math support libraries
+ * @{
+ * @addtogroup TauLabsStatistics Tau Labs statistics math support libraries
+ * @{
+ *
+ * @file       statistics.c
+ * @author     Kenn Sebesta, Copyright (C) 2015.
+ * @brief      Statistics support
+ *
+ *****************************************************************************/
+
+#include "statistics.h"
+
+#include <math.h>
+
+
+/**
+ * @brief bayes_filter Implements a bayesian testing under the effect of an
+ * actuator that can modify what is being sensed.
+ * Belief_H0 is the H0 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle has landed.
+ * Belief_H1 is the H1 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle is still flying.
+ *
+ * Inspired by Probablistic Robotics, Chap. 2.4, "Bayes Filters"
+ * @param belief_H0 The H0 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle has landed.
+ * @param belief_H1 The H1 hypothesis (also called the NULL HYPOTHESIS), e.g.
+ * the vehicle is still flying.
+ * @param p_sense_condition_H0 The probability of sensing the H0 condition, given the real condition
+ * @param p_sense_condition_H1 The probability of sensing the H1 condition, given the real condition
+ * @param p_H0_Ux_H0 The probability of state H0 staying H0 after the control is applied
+ * @param p_H0_Ux_H1 The probability of state H1 becoming H0 after the control is applied
+ * @param p_H1_Ux_H0 The probability of state H0 becoming H1 after the control is applied
+ * @param p_H1_Ux_H1 The probability of state H1 staying H1 after the control is applied
+ */
+void bayes_filter(double *belief_H0, double *belief_H1,
```

Comment:
Understood. Because it's currently unhooked to anything, I'm expecting you won't bring this over until you have a solid use for it, at which time you'll have resolved for that usage whether singles are acceptable.

The submission is to get people's brains thinking about all the ways statistics can be powerful, and give them a unit-tested library to try stuff out. I'm excited to see what people come up with.